### DISCUSSION TITLE
Thanks for amazing software :)

### DISCUSSION THREAD

**Discussion by ghost** [svsdval/video2midi#19] [Show and tell]
There are piano versions of good songs on YouTube, and most of them don't have MIDI, like https://www.youtube.com/watch?v=WDkk5MuhgJs 

I managed to convert that video to MIDI and then to MuseScore (it can import MIDI files) - https://musescore.com/user/36994868/scores/6844574/ 

There are a couple of small issues I've stumbled upon:
1) The video has consecutive keypresses without it actually being shown on the virtual "piano", so I had to move the key mapping above the piano, shift them to the left a bit (because there's a note name displayed), and "hide" the semitone keys on one of the black keys since the video doesn't use semitones at all:
![image](https://user-images.githubusercontent.com/21169548/123696784-ccd87880-d864-11eb-8538-6f08603950aa.png)

2) I haven't figured how to change the base octave, pressing "/" does nothing for me, so I had to transpose the MIDI track in LMMS after converting.

All in all, your software is very good, thanks!

**Comment by svsdval** [comment]
First of all i'm glad if my app was useful ;)
Now, about your video, you made right choice to adjust all keys on same line, here no way to determine otherwise ;)
If on other video will be semitone keys, use roll check functionality, this will reduce false triggering white keys near black keys. And you less manual editing will be needed ;)

About change base octave use this keys:
 - ]use to increase bass octave by 1 ( transpose keys by 12 tones )
 - [ use to decrease base octave by 1 ( transpose keys by 12 tones )
- The '/' is a text separator ( replacement of the OR ) for two keys [ and ]
  
If you need custom transpose by 1 tone up and down, i can implimant this

And more about this video, the video created for two pianos, first piano to bass staff, second for treble staff... no possibility to press all partitudes for single player,  i'll recommended to try to separate sheet for multiple players ;)
single per staff for: bass staff and treble staff

**Reply by ivy-rew** [reply]
maybe @Yardanico is like me using a special Keyboard layout. The '[' and ']' keys do not work are here with a German/Switzerland layout. If I switch to English (UK) I can flawlessly change the octave.
As a first simple improvement I'll try to make the 'octave' settings persistent in the .ini file saved and loaded via F2/F3. Then users can at least change the octave in the stored file and reload it from there.

**Reply by svsdval** [reply]
As variant maybe just add new buttons near octave label into settings window to adjust the octave settings ?
